include/klee/Solver.h

```diff
@@ -17,16 +17,23 @@
 #include <vector>
 
 namespace klee {
-  class ConstraintManager;
+  class ConstraintSet;
   class Expr;
   class SolverImpl;
 
+  /// Collection of meta data used by a solver
+  ///
+  struct SolverQueryMetaData {
```

Comment:
Shouldn't this be more flexible<sup>TM</sup>? I'm just thinking how it is meant to be extended? Is the idea that whoever just adds whatever field they like? But isn't that a bit inefficient for all the cases where it's not needed? Wouldn't some templating be required? Or I wouldn't mind just a `void*`. 

Also this "`SolverQueryMetaData`" seems to be only used in the TimingSolver, so why is it a full `Solver.h`thing? Maybe the best option is to call it TimingMetaData and just use it for the TimingSolver and leave a full Solver chain metadata thing for a later PR?